### PR TITLE
Update release tool to check next_release

### PR DESCRIPTION
**Pull request by sebastienvas**
Also adds a requirement to specify the sha to create the release from.
Now a release can be tagged on istio with:
```
githubctl --token_file=<github token file> \
    --op=uploadArtifacts --base_branch=<release branch or master> \
    --next_release=0.2.2 --ref_sha=<sha>
```
```release-note
none
```

### CODE REVIEW

File: toolbox/githubctl/main.go [hunk 1]
```diff
@@ -267,7 +270,8 @@ func main() {
 			log.Printf("Error during UpdateIstioVersionAfterReleaseTagsMadeOnDeps: %v\n", err)
 		}
 	case "uploadArtifacts":
-		if err := CreateIstioReleaseUploadArtifacts(); err != nil {
+		assertNotEmpty("ref_sha", refSHA)
```
Comment (chxchx) [inline toolbox/githubctl/main.go]: This works fine but we might want to move the assertion into subroutine (see line 216 217) to be consistent with the rest of the code.

File: toolbox/util/githubClient.go [hunk 2]
```diff
@@ -336,9 +336,12 @@ func (g GithubClient) CreateAnnotatedTag(repo, tag, sha, msg string) error {
 
 // CreateReleaseUploadArchives creates a release given release tag and
 // upload all files in archiveDir as assets of this release
-func (g GithubClient) CreateReleaseUploadArchives(repo, releaseTag, archiveDir string) error {
+func (g GithubClient) CreateReleaseUploadArchives(repo, releaseTag, sha, archiveDir string) error {
```
Comment (chxchx) [inline toolbox/util/githubClient.go:339]: Also update the function comment to explain what `sha` is used for

File: toolbox/githubctl/main.go [hunk 3]
```diff
@@ -212,10 +212,13 @@ func UpdateIstioVersionAfterReleaseTagsMadeOnDeps() error {
 }
 
 // CreateIstioReleaseUploadArtifacts creates a release on istio and uploads dependent artifacts
-func CreateIstioReleaseUploadArtifacts() error {
+func CreateIstioReleaseUploadArtifacts(refSHA string) error {
 	assertNotEmpty("base_branch", baseBranch)
 	assertNotEmpty("next_release", nextRelease)
 	releaseTag, err := getReleaseTag()
+	if releaseTag == *nextRelease {
+		return fmt.Errorf("Next Release tag needs to be greater than the current release.")
```
Comment (chxchx) [inline toolbox/githubctl/main.go]: error should start with lower case letter

Comment (sebastienvas) [reply toolbox/githubctl/main.go]: Done

File: toolbox/githubctl/main.go (hunk 1 above)
Comment (sebastienvas) [reply toolbox/githubctl/main.go]: Done

File: toolbox/util/githubClient.go (hunk 2 above)
Comment (sebastienvas) [reply toolbox/util/githubClient.go:339]: Done

**Comment by sebastienvas** [comment]
PTAL

**Pull request merged**